cylc/cylc-uiserver
# Clean gui info files

closes: https://github.com/cylc/cylc-uiserver/issues/416

This fixes a bug whereby, if an existing ui-server is killed with a `kill -9` or with a system crash, jupyter server info files (which serve as a contact file) can be left behind. This is problematic as our logic (added in https://github.com/cylc/cylc-uiserver/pull/370) can select the broken server for re-use which results in the gui opening at an error page, rather than the cylc gui.

**How to test**
For reviewers to test (on master to reproduce the bug)...
- clean your info files dir (~/.cylc/uiserver/info-files)
- open a uiserver with `cylc gui`
- look at info files dir, you should have two files (one .json and a html open file), take the pid from this filename
- kill the process (kill -9 <pid>)
- run another gui `cylc gui` (don't use the `--new` option)
- you should get an error page instead of the cylc gui

Test again on this branch and the bug should be fixed, with a prompt asking you if you wish to remove the files (either response should result in a gui being opened correctly)
 
Note that this branch also changes the logic - the html open file is now no longer written to in this process, so functional testing of various gui openings (e.g. with/without `--new` and `workflow_id`) would be prudent.

- [x] If this is a bug fix, PR should be raised against the relevant branch.
